hypothesis-python/src/hypothesis/extra/github_actions.py

```diff
@@ -0,0 +1,156 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Copyright the Hypothesis Authors.
+# Individual contributors are listed in AUTHORS.rst and the git log.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+import shutil
+import tempfile
+from os import getenv
+
+import requests
+
+from hypothesis.configuration import mkdir_p, storage_directory
+from hypothesis.database import DirectoryBasedExampleDatabase
+
+
+class GitHubArtifactDatabase(DirectoryBasedExampleDatabase):
+    """
+    A directory-based database loaded from a GitHub Actions artifact.
+
+    This is useful for sharing example databases between CI runs and developers, allowing the latter
+    to get read-only access to the former. In most cases, this will be used through the
+    :class:`~hypothesis.database.MultiplexedDatabase`, by combining a local directory-based database
+    with this one. For example:
+
+    .. code-block:: python
+
+        local = DirectoryBasedExampleDatabase(".hypothesis/examples")
+        shared = GitHubArtifactDatabase("user", "repo")
+
+        settings.register_profile("ci", database=local)
+        settings.register_profile("dev", database=MultiplexedDatabase(local, shared))
+        settings.load_profile("ci" if os.environ.get("CI") else "dev")
+
+    This, combined with a CI workflow that syncs its local folder to an artifact, will
+    allow developers to also get access to the examples generated by CI.
+
+    .. note::
+        This database is read-only, and will not upload any examples to the artifact. This is to
+
+
+    For mono-repo support, you can provide an unique `artifact_name` (e.g. `hypofuzz-example-db-branch`).
+    """
+
+    def __init__(
+        self,
+        owner: str,
+        repo: str,
+        artifact_name: str = "hypofuzz-example-db",
+        path: str = storage_directory("ci"),
```

Comment:
```suggestion
        path: str = storage_directory("gh-artifacts"),
```

`ci` is a bit too generic of a name here.